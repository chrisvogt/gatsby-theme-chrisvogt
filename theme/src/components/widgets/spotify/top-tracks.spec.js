/** @jsx jsx */
import renderer from 'react-test-renderer'
import { jsx } from 'theme-ui'
import TopTracks from './top-tracks'

describe('TopTracks Component', () => {
  const mockTracks = [
    {
      id: '1',
      name: 'Song One',
      artists: ['Artist One', 'Artist Two'],
      albumImages: [
        { url: 'http://example.com/large.jpg', width: 640 },
        { url: 'http://example.com/medium.jpg', width: 300 },
        { url: 'http://example.com/small.jpg', width: 64 }
      ],
      spotifyURL: 'http://spotify.com/track1'
    },
    {
      id: '2',
      name: 'Song Two',
      artists: ['Artist Three'],
      albumImages: [
        { url: 'http://example.com/medium2.jpg', width: 300 },
        { url: 'http://example.com/small2.jpg', width: 64 }
      ],
      spotifyURL: 'http://spotify.com/track2'
    }
  ]

  it('renders correctly with tracks', () => {
    const tree = renderer.create(<TopTracks isLoading={false} tracks={mockTracks} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when loading', () => {
    const tree = renderer.create(<TopTracks isLoading={true} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles tracks without a 300px image gracefully', () => {
    const incompleteTracks = [
      {
        id: '3',
        name: 'Song Three',
        artists: ['Artist Four'],
        albumImages: [{ url: 'http://example.com/small3.jpg', width: 64 }],
        spotifyURL: 'http://spotify.com/track3'
      }
    ]
    const tree = renderer.create(<TopTracks isLoading={false} tracks={incompleteTracks} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with no tracks', () => {
    const tree = renderer.create(<TopTracks isLoading={false} tracks={[]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
