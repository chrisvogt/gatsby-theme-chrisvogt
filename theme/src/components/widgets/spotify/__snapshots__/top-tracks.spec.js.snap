// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TopTracks Component handles tracks without a 300px image gracefully 1`] = `
<div
  className="css-sak8bt"
>
  <div
    className="css-ccroti"
  >
    <h3
      className="css-9nh3l5"
    >
      Top Tracks
    </h3>
  </div>
  <p
    className="css-1qwovgy"
  >
    My 12 most-played tracks over the last 4 weeks.
  </p>
  <div
    data-testid="media-item-grid"
  />
</div>
`;

exports[`TopTracks Component renders correctly when loading 1`] = `
<div
  className="css-sak8bt"
>
  <div
    className="css-ccroti"
  >
    <h3
      className="css-9nh3l5"
    >
      Top Tracks
    </h3>
  </div>
  <p
    className="css-1qwovgy"
  >
    My 12 most-played tracks over the last 4 weeks.
  </p>
  <div
    data-testid="media-item-grid"
  />
</div>
`;

exports[`TopTracks Component renders correctly with no tracks 1`] = `
<div
  className="css-sak8bt"
>
  <div
    className="css-ccroti"
  >
    <h3
      className="css-9nh3l5"
    >
      Top Tracks
    </h3>
  </div>
  <p
    className="css-1qwovgy"
  >
    My 12 most-played tracks over the last 4 weeks.
  </p>
  <div
    data-testid="media-item-grid"
  />
</div>
`;

exports[`TopTracks Component renders correctly with tracks 1`] = `
<div
  className="css-sak8bt"
>
  <div
    className="css-ccroti"
  >
    <h3
      className="css-9nh3l5"
    >
      Top Tracks
    </h3>
  </div>
  <p
    className="css-1qwovgy"
  >
    My 12 most-played tracks over the last 4 weeks.
  </p>
  <div
    data-testid="media-item-grid"
  />
</div>
`;
